src/jazzlights/effect/creatures.cpp

```diff
@@ -95,18 +91,42 @@ void KnownCreatures::AddCreature(uint32_t color, Milliseconds lastHeard, int rss
     Creature new_creature;
     new_creature.color = color;
     new_creature.lastHeard = lastHeard;
-    new_creature.rssi = rssi;
+    new_creature.lastHeardRssi = rssi;
+    new_creature.effectiveRssi = rssi;
+    new_creature.lastHeardNearby =
+        (CreatureIntensity(new_creature) >= Creatures::kCloseCreatureIntensityThresh) ? lastHeard : 0;
     creatures_.push_back(new_creature);
   }
   std::sort(creatures_.begin(), creatures_.end(),
             [](Creature a, Creature b) { return ColorHash(a.color) > ColorHash(b.color); });
 }
 
+void KnownCreatures::update() {
+  Milliseconds currentTime = timeMillis();
+  for (Creature& creature : creatures_) {
+    if (creature.lastHeard > 0 && currentTime - creature.lastHeard > KnownCreatures::kRssiDecayDelayMs) {
+      // If we haven't heard from this creature in kRssiDecayDelayMs, decay the RSSI by kRssiDecayFactor dBm every
+      // second.
+      creature.effectiveRssi =
+          creature.lastHeardRssi -
+          (KnownCreatures::kRssiDecayFactor * (currentTime - creature.lastHeard + KnownCreatures::kRssiDecayDelayMs)) /
```

Comment:
Should this `+` be a `-` in order to have the decay start at 0?

```suggestion
          (KnownCreatures::kRssiDecayFactor * (currentTime - creature.lastHeard - KnownCreatures::kRssiDecayDelayMs)) /
```